skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,322 @@
+  private val noUi: Boolean = false,
+) : CircuitComponent {
+  override val fileSuffix = "Presenter"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /*
+     Generate Circuit Presenter class, eg:
+
+     class FeaturePresenter @AssistedInject constructor(
+       @Assisted
+       private val screen: {Feature}Screen,
+       @Assisted
+       private val navigator: Navigator,
+     ) : Presenter<ggScreen.State> {
+       @Composable
+       override fun present(): FeatureScreen.State {
+         val scope = rememberStableCoroutineScope()
+         return FeatureScreen.State() { event ->
+         }
+       }
+
+       @AssistedFactory
+       @CircuitInject(
+         FeatureScreen::class,
+         UserScope::class,
+       )
+       interface Factory {
+         fun create(screen: ggScreen, navigator: Navigator): FeaturePresenter {
+         }
+       }
+     }
+    */
+    val screenClassName = "${className}Screen"
+    val presenterClassName = "${className}Presenter"
+    val assistedInject = ClassName("dagger.assisted", "AssistedInject")
+    val assisted = ClassName("dagger.assisted", "Assisted")
+    val assistedFactory = ClassName("dagger.assisted", "AssistedFactory")
+
+    val factoryBuilder =
+      TypeSpec.interfaceBuilder("Factory").apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedFactory)
+        }
+      }
+
+    val constructorBuilder =
+      FunSpec.constructorBuilder().apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedInject)
+        }
+        addParameter(
+          ParameterSpec.builder("screen", ClassName("", screenClassName))
+            .apply {
+              if (useAssistedInjection) {
+                addAnnotation(assisted)
+              }
+            }
+            .build()
+        )
+        addParameter(
+          ParameterSpec.builder("navigator", ClassName("com.slack.circuit.runtime", "Navigator"))
+            .apply {
+              if (useAssistedInjection) {
+                addAnnotation(assisted)
+              }
+            }
+            .build()
+        )
+      }
+
+    val presenterClass =
+      TypeSpec.classBuilder(presenterClassName).apply {
+        if (noUi) {
+          addKdoc(
+            """
+         TODO (remove): This Circuit [Presenter] was generated without UI or [CircuitInject], and can be
+         used with legacy views via `by circuitState()` in your Fragment or Activity.
+          """
+              .trimIndent()
+          )
+        }
+        primaryConstructor(constructorBuilder.build())
+          .addSuperinterface(
+            ClassName("com.slack.circuit.runtime.presenter", "Presenter")
+              .parameterizedBy(ClassName(packageName, "${className}Screen.State"))
+          )
+          .addProperty(
+            PropertySpec.builder("screen", ClassName("", "${className}Screen"))
+              .addModifiers(KModifier.PRIVATE)
```

Comment:
These modifiers will get stripped by your post-processing to remove `public `. Can we refine it to be more specific in which declarations it strips the prefix from?